src/components/Overview/Overview.jsx

```diff
@@ -22,8 +22,8 @@ const Overview = ({productId, productDetails, ratings, numOfReviews, setQuantity
     })
   }
 
-  useEffect(() => {getStyles(productId)}, [])
-  useEffect(() => {setCurrentStyle(styles[0])}, [styles])
+  useEffect(() => {getStyles(productId)}, [productId])
+  useEffect(() => {setCurrentStyle(styles[0])}, [styles, productId])
```

Comment:
I think for line 26, we don't need productId as a factor. Does the page not render correctly without productId? if so, please ignore my comment.